### PR TITLE
purple: Implement prplcb_close_request for request_input

### PR DESCRIPTION
**Pull request by dequis**
Followup to #22 and #23

Adds more stuff to the `request_input_data` struct so that `prplcb_close_request` has more context to work with.

Tested and valgrind'd. Just needs some review for nitpicks.

### CODE REVIEW

File: protocols/purple/purple.c [hunk 1]
```diff
@@ -1074,19 +1077,21 @@ static void *prplcb_request_action(const char *title, const char *primary, const
  */
 static void prplcb_close_request(PurpleRequestType type, void *data)
 {
+	if (!data) {
+		return;
+	}
+
 	if (type == PURPLE_REQUEST_ACTION) {
```
Comment (jgeboski) [inline protocols/purple/purple.c]: `switch` it up?

**Comment by jgeboski** [comment]
Looks good pending testing.

File: protocols/purple/purple.c (hunk 1 above)
Comment (dequis) [inline protocols/purple/purple.c]: At heart I'm still a python programmer. `switch` does not exist.

Comment (dequis) [inline protocols/purple/purple.c]: Nah scratch that, switch looks more readable here.

**Comment by dequis** [comment]
Squashed and rephrased commit message

**Pull request merged**